sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -314,6 +314,8 @@ class SQLListener(conf: SparkConf) extends SparkListener with Logging {
       if (executionUIData.isFailed) {
         failedExecutions += executionUIData
         trimExecutionsIfNecessary(failedExecutions)
+      } else if (executionUIData.jobs.isEmpty && executionUIData.accumulatorMetrics.isEmpty) {
```

Comment:
I prefer to not do this. Otherwise, such query will disappear after it finishes on UI and it seems pretty weird.